### ISSUE TITLE
Wide alignment preview in the editor is not accurate

### ISSUE THREAD

**Issue by afercia** [WordPress/twentynineteen#667]
To reproduce:

- enter some content in a paragraph
- add an Image block with a relatively small image, to make things easier to test:

![screenshot 2018-11-25 at 17 19 04](https://user-images.githubusercontent.com/1682452/48981526-d9a5f600-f0d6-11e8-9693-960c4a138742.png)

- click the Preview button to see the front-end
- as expected, the preview in the editor closely matches the front-end
- now set the image to a "wide" alignment:

![screenshot 2018-11-25 at 17 28 56](https://user-images.githubusercontent.com/1682452/48981598-d6f7d080-f0d7-11e8-8fb0-b76ac8640717.png)

- notice the image is no larger than the paragraph content above
- preview in the frontend:
- the image is larger than the paragraph content:

![screenshot 2018-11-25 at 17 19 29](https://user-images.githubusercontent.com/1682452/48981563-4f11c680-f0d7-11e8-8584-1666d4e6c713.png)

- in the editor, close the Settings sidebar
- now the image is larger than the paragraph content also in the editor and matches the real layou tin the front-end:

![screenshot 2018-11-25 at 17 29 09](https://user-images.githubusercontent.com/1682452/48981603-00186100-f0d8-11e8-8c68-1bcf82a29ec5.png)

Expected: the view in the editor to match the layout in the front-end, regardless of the sidebar state.

**Comment by joyously** [comment]
100% is variable. It would be the same with full width, I bet.
And I think that is reasonable, because the preview could be a different sized window. And the final result could be any sized window or phone.

**Comment by afercia** [comment]
>  It would be the same with full width, I bet.

No, full width is OK. It always takes all the available space regardless of the sidebar state.

I don't think it's reasonable. The problem is the editor view UI is lying to users, as it doesn't matches the front-end. This is even more evident with a large image. If you try with a large image there's really no difference between "normal" and "wide" (with the Sidebar open).

The wide alignment in the editor should use some `calc()` to display a wide image left aligned and extended on the right, in the same way it uses some `calc()` for `.wp-block[data-align="full"]`.

**Comment by joyously** [comment]
It _does_ match the front end, but the front end does not have a sidebar. So 100% can be different, depending on the window. The style shouldn't be different just because it's in the editor. What happens if you put a wide thing in a column?

**Comment by kjellr** [comment]
Thanks for the report, @afercia. As you noted, the issue goes away when you turn off the settings sidebar. The wide alignment has been designed and implemented to behave this way. When you have the settings sidebar open in your example, the "viewport" for the content is less wide, and therefore drops to use narrower breakpoint styles. The blocks behave as if you'd narrowed your entire browser window by the width of the sidebar(s). 

![width](https://user-images.githubusercontent.com/1202812/48982172-e6155900-f0ac-11e8-8fea-291ad1174197.gif)

Depending  on your screen size, Gutenberg behaves this way by default too: 

![width2](https://user-images.githubusercontent.com/1202812/48982236-e3ffca00-f0ad-11e8-87b1-f9ffe60fef69.gif)

Since this is working as intended, I'm going to close this for now, but we can still continue discussing if necessary.

**Comment by afercia** [comment]
> The style shouldn't be different just because it's in the editor.

For the full width alignment, it is already different 🙂 And it's reasonable, otherwise the full alignment in the editor would just be broken.

*.alignfull in the front-end:
```
left: calc( -12.5% - 75px);
width: calc( 125% + 150px);
max-width: calc( 125% + 150px);
```

.wp-block[data-align="full"] in the editor:
```
left: calc( -12.5% - 14px);
width: calc( 125% + 116px);	
max-width: calc( 125% + 115px);
```

@kjellr I kindly disagree 🙂 
In the front-end, the wide image is always larger than the paragraph content even at smaller viewports, until it goes fully responsive under 768. This is not the case in the editor, so the views don't match.

**Comment by kjellr** [comment]
Thanks for the persistence, @afercia. 

I do see that when the sidebars are active, alignwide elements are not exactly 1:1 compared to their  front-end equivilents at the same width: 

![width2](https://user-images.githubusercontent.com/1202812/48982714-26c4a080-f0b4-11e8-9906-dfe113b4f052.gif)

This is _not_ a problem when comparing two full-screen views between the front and back end: 

![width3](https://user-images.githubusercontent.com/1202812/48982725-3c39ca80-f0b4-11e8-8c03-cba54ae8ef6e.gif)

I'll re-open for now.
